### ISSUE TITLE
Bug: MonitorWake non functional after update to Windows 11 24H2

### ISSUE THREAD

**Issue by EPiSKiNG** [hass-agent/HASS.Agent#182]
I have a surface pro 7 without a keyboard attached to it and not connected to any external monitor. When I trigger the monitorwake command, I see the cursor arrow move up, so I know the command is arriving to the device. However, when the screen is off (and I still have network because I can ping), the up arrow does not wake the monitor.

I assume this must be some change in Windows 24H2, as it was working properly prior to my in-place upgrade to 24H2.

Thoughts?

**Comment by amadeo-alex** [comment]
Unfortunately yes. What you're facing is most likely called "modern sleep" aka "s0ix" sleep. Bane of my sanity for a long time.
Allow me to post a screenshot of explanation I did on discord:
![2zD59p](https://github.com/user-attachments/assets/d9b64574-f5bc-498f-8679-3730a0fb6926)

The only strange thing is that it happened to you after and upgrade.
Did you have the keyboard attached before? Asking since some USB devices can keep the device alive and prevent "modern sleep".

**Comment by EPiSKiNG** [comment]
No, the keyboard was not attached before or after the upgrade... Looks like I may have to figure out a way to disable modern sleep or disable the lock screen and do a sleep command to turn it off and a wake on wlan to turn it back on...?

Fiddlesticks! Lol

**Comment by amadeo-alex** [comment]
@EPiSKiNG if you'd like, please try this test version https://github.com/amadeo-alex/HASS.Agent/releases/tag/2.1.1-beta2-monitorsleep

Installation and configuration instructions are in the release - please note, following those will prevent your device from sleeping :D

**Comment by AJolly** [comment]
btw, I deal with this sorta thing by first triggering WOL then monwake.  Also I have it set to not use the up key, but rather keycode 251

**Comment by EPiSKiNG** [comment]
Got it set up, verified with powercfg /requests that it is functional.

Triggered the command via MQTT, and the screens went dark.  Touched the screen and it came back on.  Waited about 5 minutes then triggered it again, and the screen didn't go dark... Perhaps because I have video playing on the screen?  Closed the browser (home assistant dashboard with frigate card), and re-ran the command then it did indeed turn the screen off.  I think there is some settings in the power configuration that disables sleep/monitors going off when media is playing?

For me this is all about power consumption.  I'm using a Surface Pro 7.  When on the HA dashboard with frigate card running, I'm pulling around 18w.  After triggering "monitorsleeppowerplan" (after closing the browser, the consumption drops to 7.5w, which is great, however, sending the device to "Hibernate" drops the consumption down to less than 1w... That in combination with WoWLAN seems like the better option for me at this time.

Thank you for the development and I look forward to your future releases!

**Comment by EPiSKiNG** [comment]
I recently re-deployed my surface device with standard Windows 11 image.  I wasn't able to get the monitorsleeppowerplan to function, and instead had to do the following registry entry to disable Modern Standby, then I was able to use the monitorsleep command in conjunction with monitorwake.

CMD as Admin:
`reg add HKLM\System\CurrentControlSet\Control\Power /v PlatformAoAcOverride /t REG_DWORD /d 0`
